onnx/reference/ops/op_rotary_embedding.py

```diff
@@ -84,6 +86,7 @@ def rotary_embedding(
     else:
         x_rotate = np.concatenate((real, imag), axis=-1)
     output = np.concatenate((x_rotate, x_not_rotate), axis=-1)
+    output = np.transpose(output, (0, 2, 1, 3))
```

Comment:
Seems like this transpose can erroneously happen for the input=3D case also?

Reply:
Updated

Reply:
Was there any test cases to catch this?

Reply:
Good question. It's a bit tricky. Let's discuss this separately, I am resolving this in order to merge this PR. The short answer is: what would you compare it against? The same mistake exists in both the reference-implementation and the function-body (otherwise, I think it would have been caught). There is no third source of truth ... the generated data acts as a source of truth for other onnx implementations (like ORT) ... which would fail, I suppose when something like ORT is tested.